### ISSUE TITLE
setBindings() does not overwrite existing bindings

### ISSUE THREAD

**Issue by deanhaleem** [pinojs/pino#1418]
Hello!

While using `setBindings()` and referring to the docs comment in the [types](https://github.com/pinojs/pino/blob/master/pino.d.ts#L107-L112), I expected it to at least overwrite existing fields. However, it ends up duplicating the field in the log.

Is this the intended behavior? Based on [this unit test](https://github.com/pinojs/pino/blob/master/test/basic.test.js#L41-L45), I didn't think it was the desired behavior. But since `bindings()` uses `JSON.parse()`, the unit test doesn't catch that `pino.chindings` includes duplicate keys.

### Steps to reproduce
```js
const pino = require('pino')

const logger = pino({ name: 'basicTest', level: 'info', base: { foo: 'bar' } })

logger.info('Hello World')

logger.setBindings({ foo: 'baz' })

logger.info('Hello World')

```

Running this outputs:
```
{"level":30,"time":1651029373867,"foo":"bar","name":"basicTest","msg":"Hello World"}
{"level":30,"time":1651029373867,"foo":"bar","name":"basicTest","foo":"baz","msg":"Hello World"}
```

Notice the duplicate `foo` field.

**Comment by mcollina** [comment]
This was a design choice for performance reasons . I don't see us changing this, would you like to send a PR to the did to clarify?

**Comment by jsumners** [comment]
Resolved by #1419.
